src/postman/transformers/request.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  HttpParamStyles,
+  IHttpHeaderParam,
+  IHttpOperationRequestBody,
+  IHttpPathParam,
+  IHttpQueryParam,
+  IMediaTypeContent,
+} from '@stoplight/types';
+import { JSONSchema4 } from 'json-schema';
+import { FormParam, HeaderDefinition, PropertyList, QueryParam, RequestBody } from 'postman-collection';
+import { transformDescriptionDefinition, transformPostmanTemplate, transformValueToHttpParam } from '../util';
+
+export function transformQueryParam(queryParam: QueryParam): IHttpQueryParam {
+  return {
+    name: queryParam.key as string, // no key no game
+    style: HttpParamStyles.Form,
+    ...(queryParam.value ? transformValueToHttpParam(queryParam.value) : undefined),
+  };
+}
+
+export function transformHeader(header: HeaderDefinition): IHttpHeaderParam {
+  return {
+    name: header.key as string,
+    style: HttpParamStyles.Simple,
+    ...(header.value ? transformValueToHttpParam(header.value) : undefined),
+  };
+}
+
+export function transformPathParams(segments: string[]): IHttpPathParam[] {
+  return segments.reduce<IHttpHeaderParam[]>((params, segment) => {
+    if (segment.startsWith(':')) {
+      params.push({
+        name: segment.substring(1),
+        style: HttpParamStyles.Simple,
+      });
+    }
+
+    return params;
+  }, []);
+}
+
+export function transformBody(body: RequestBody, mediaType?: string): IHttpOperationRequestBody | undefined {
+  switch (body.mode) {
+    case 'raw':
+      if (!body.raw) return;
+      return { contents: [transformRawBody(body.raw, mediaType)] };
+
+    case 'formdata':
+      if (!body.formdata) return;
+      return {
+        contents: [transformParamsBody<FormParam>(body.formdata, mediaType || 'multipart/form-data')],
+      };
+
+    case 'urlencoded':
+      if (!body.urlencoded) return;
+      return {
+        contents: [transformParamsBody<QueryParam>(body.urlencoded, mediaType || 'application/x-www-form-urlencoded')],
+      };
+  }
+
+  return;
```

Comment:
1. since http-spec is used as a library we should probably establish a channel for communicating warnings. Simple `console.warn` could make too much noise (thinking about platform logs). On the other hand, if there is noise, we should implement missing mode.
2. the action if there is no corresponding-to-mode key is skipping entire body. It is hard to say whether such case is a result of error or expected case, but our reaction seems to be OK